storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -103,3 +103,52 @@ func (d *DotGit) Idxfile() (fs.FS, string, error) {
 
 	return nil, "", ErrIdxNotFound
 }
+
+func (d *DotGit) Objectfiles() (fs.FS, []core.Hash, error) {
+	dotGitobjcts := d.fs.Join(d.path, "objects")
+	files, err := d.fs.ReadDir(dotGitobjcts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var objDirs []string
+	reDir, _ := regexp.Compile("[a-z0-9]{2}")
```

Comment:
The regexp is wrong in several ways:

- wrong characters: characters like `g` or `z` are allowed, while the only characters allowed for the directory names are 0-9 and a-f.

- The refgexp matches non valid directory names like '000' or 'pack' (as 'ac' is valid). What is the point of having a regexp then?